src/editors/selectEditor.js

```diff
@@ -15,233 +15,288 @@ import { KEY_CODES } from './../helpers/unicode';
+   */
+  open() {
+    this._opened = true;
+    this.refreshDimensions();
+    this.select.style.display = '';
+    this.instance.addHook('beforeKeyDown', () => this.onBeforeKeyDown());
+  }
 
-  if (Array.isArray(optionsToPrepare)) {
-    for (let i = 0, len = optionsToPrepare.length; i < len; i++) {
-      preparedOptions[optionsToPrepare[i]] = optionsToPrepare[i];
-    }
-  } else if (typeof optionsToPrepare === 'object') {
-    preparedOptions = optionsToPrepare;
+  /**
+   * Closes the editor.
+   */
+  close() {
+    this._opened = false;
+    this.select.style.display = 'none';
+    this.instance.removeHook('beforeKeyDown', () => this.onBeforeKeyDown());
   }
 
-  return preparedOptions;
+  /**
+   * Sets focus state on the select element.
+   */
+  focus() {
+    this.select.focus();
+  }
 
-};
+  /**
+   * Binds hooks to refresh editor's size after scrolling of the viewport or resizing of columns/rows.
+   *
+   * @private
+   */
+  registerHooks() {
+    this.instance.addHook('afterScrollHorizontally', () => this.refreshDimensions());
+    this.instance.addHook('afterScrollVertically', () => this.refreshDimensions());
+    this.instance.addHook('afterColumnResize', () => this.refreshDimensions());
+    this.instance.addHook('afterRowResize', () => this.refreshDimensions());
+  }
 
-SelectEditor.prototype.getValue = function() {
-  return this.select.value;
-};
+  /**
+   * Prepares a list of available options.
+   */
+  prepare(...args) {
+    super.prepare(...args);
 
-SelectEditor.prototype.setValue = function(value) {
-  this.select.value = value;
-};
+    const selectOptions = this.cellProperties.selectOptions;
+    let options;
 
-const onBeforeKeyDown = function(event) {
-  const instance = this;
-  const editor = instance.getActiveEditor();
-  const previousOptionIndex = editor.select.selectedIndex - 1;
-  const nextOptionIndex = editor.select.selectedIndex + 1;
+    if (typeof selectOptions === 'function') {
+      options = this.prepareOptions(selectOptions(this.row, this.col, this.prop));
+    } else {
+      options = this.prepareOptions(selectOptions);
+    }
 
-  switch (event.keyCode) {
-    case KEY_CODES.ARROW_UP:
-      if (previousOptionIndex >= 0) {
-        editor.select[previousOptionIndex].selected = true;
-      }
+    empty(this.select);
 
-      stopImmediatePropagation(event);
-      event.preventDefault();
-      break;
+    objectEach(options, (value, key) => {
+      const optionElement = document.createElement('OPTION');
+      optionElement.value = key;
 
-    case KEY_CODES.ARROW_DOWN:
-      if (nextOptionIndex <= editor.select.length - 1) {
-        editor.select[nextOptionIndex].selected = true;
+      fastInnerHTML(optionElement, value);
+      this.select.appendChild(optionElement);
+    });
+  }
+
+  /**
+   * Creates consistent list of available options.
+   *
+   * @param {Array|Object} optionsToPrepare
+   * @returns {Object}
+   */
+  prepareOptions(optionsToPrepare) {
```

Comment:
Shouldn't this method be private?

Same with `refreshValue`, `refreshDimensions`.